cmd/ulxly/ulxly.go

```diff
@@ -126,6 +182,294 @@ var DepositsCmd = &cobra.Command{
+			},
+			// Value:     value,
+			From:      fromAddress,
+			Context:   ctx,
+			GasLimit:  gasLimit,
+			GasPrice:  gasPrice,
+			GasFeeCap: nil,
+			GasTipCap: nil,
+		}
+
+		/////////////////////////////////////
+		// This section calls the bridge service RPC URL to get the deposits data and parses them to the correct formats.
+		bridgeServiceDepositsEndpoint := fmt.Sprintf("%s/bridges/%s", *ulxlyInputArgs.BridgeServiceRPCURL, *ulxlyInputArgs.ClaimAddress)
+		reqBridgeDeposits, err := http.Get(bridgeServiceDepositsEndpoint)
+		if err != nil {
+			log.Error().Err(err)
+		}
+		bodyBridgeDeposit, err := io.ReadAll(reqBridgeDeposits.Body) // Response body is []byte
+		if err != nil {
+			log.Error().Err(err)
+		}
+		var bridgeDeposit BridgeDeposits
+		err = json.Unmarshal(bodyBridgeDeposit, &bridgeDeposit) // Parse []byte to go struct pointer, and shadow err variable
+		if err != nil {
+			log.Error().Err(err).Msg("Can not unmarshal JSON")
+			return err
+		}
+
+		var originAddress common.Address
+		globalIndex := new(big.Int)
+		amount := new(big.Int)
+		var metadata []byte
+
+		intClaimIndex, _ := strconv.Atoi(*ulxlyInputArgs.ClaimIndex) // Convert deposit_cnt to int
+		for index, deposit := range bridgeDeposit.Deposit {
+			intDepositCnt, _ := strconv.Atoi(deposit.DepositCnt) // Convert deposit_cnt to int
+			if intDepositCnt == intClaimIndex {                  // deposit_cnt must match the user's input value
+				if !bridgeDeposit.Deposit[index].ReadyForClaim {
+					log.Error().Msg("The claim transaction is not yet ready to be claimed. Try again in a few blocks.")
+					return nil
+				} else if bridgeDeposit.Deposit[index].ClaimTxHash != "" {
+					fmt.Printf("The claim transaction has already been claimed at %s.", bridgeDeposit.Deposit[index].ClaimTxHash)
+					return nil
+				}
+				originAddress = common.HexToAddress(bridgeDeposit.Deposit[index].OrigAddr)
+				globalIndex.SetString(bridgeDeposit.Deposit[index].GlobalIndex, 10)
+				amount.SetString(bridgeDeposit.Deposit[index].Amount, 10)
+				metadata = common.Hex2Bytes(bridgeDeposit.Deposit[index].Metadata)
+				break
+			}
+		}
+		defer reqBridgeDeposits.Body.Close()
+		/////////////////////////////////////
+
+		bridgeAssetTxn, err := bridgeV2.ClaimAsset(tops, merkleProofArray, rollupMerkleProofArray, globalIndex, [32]byte(mainExitRoot), [32]byte(rollupExitRoot), 0, originAddress, 1, toAddress, amount, metadata)
+		if err != nil {
+			return err
+		}
+
+		// Wait for the transaction to be mined
+		for {
+			r, err := client.TransactionReceipt(ctx, bridgeAssetTxn.Hash())
+			if err != nil {
+				if err.Error() != "not found" {
+					log.Error().Err(err)
+				}
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			if r.Status != 0 {
+				fmt.Printf("Claim Transaction Successful: %s\n", r.TxHash)
+				break
+			} else if r.Status == 0 {
+				fmt.Printf("Claim Transaction Failed: %s\n", r.TxHash)
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
```

Comment:
Worth using a library or making a function since you're doing this twice. E.g. we use this in a few places i think? `github.com/cenkalti/backoff/v4`